lib/salus/report.rb

```diff
@@ -229,9 +231,28 @@ def publish_report(directive)
       end
     end
 
+    def satisfies_filter?(directive)
+      return true if @report_filter == 'all'
+      return false if @report_filter == 'none'
+
+      # Split the filter string only by the first colon
+      recovered_values = @report_filter.split(':', 2)
+      filter_key = recovered_values[0]
+      filter_value = recovered_values[1]
```

Comment:
`satisfies_filter?` needs to be refactored.
Right now it is invoked once for each uri, but things like `filter_key` and `filter_value` only need to be calculated once.
This also causes an issue if there is a poorly formatted report filter, because if there are 5 report uris, then the code below will print out this error 5 times.
```
Could not send Salus report: (Salus::Report::ExportReportError: Poorly formatted report filter found. Filter key and pattern must be non-empty strings)
```

and maybe you don't need to call `satisfies_filter?` if `@report_filter` is `all` or `none`.